py-polars/src/series/buffers.rs

```diff
@@ -245,11 +245,13 @@ impl PySeries {
 
 unsafe fn from_buffer_impl<T: NativeType>(
     pointer: usize,
+    offset: usize,
     length: usize,
     owner: Py<PyAny>,
 ) -> Box<dyn Array> {
     let pointer = pointer as *const T;
-    let slice = unsafe { std::slice::from_raw_parts(pointer, length) };
+    let slice = unsafe { std::slice::from_raw_parts(pointer, length + offset) };
```

Comment:
We can use offset to increment the pointer directly.

Reply:
Ah yeah I tried that but I didn't know about the handy `offset method`. Fixed!